library/headers.inc.php

```diff
@@ -42,43 +46,66 @@ function include_css_library($path)
     Strings equate to the directory name in the /assets/ directory.   
 */
 function call_required_libraries($library_array){
-/* First checking for any library in a directory matching the string "/jquery-min-/".
- * When one is found, use that value in the URL string, then add "index.js"
- */
-    foreach($library_array as $v){ 
-    if (preg_match("/jquery-min-/",$v)) {?>
-    <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path'].$v ; ?>/index.js"></script>    
-    <?php 
-    } }
+    /* First checking for any library in a directory matching the string "/jquery-min-/".
+     * When one is found, use that value in the URL string, then add "index.js"
+     * In order to decrease the chance of including and library calling problems, some measures were taken:
+     * - The array in the argument is completely converted to lower-case, meaning that there is not any sensitivity between jQuery & jquery or fancyBox & fancybox
+     * - Some libraries can be called with or without "-". For example: "font-awesome" and "fontawesome" => those 2 are the same thing
+     */
+    
+    $library_array = array_map('strtolower', $library_array);
```

Comment:
This forces lowercase names.  That breaks things on Linux if there is a difference.
We could accept that our standard is to do such, but I am not sure if it should be forced.  
@tmccormi 
@teryhill 
@Trodrige 
@yehster

Reply:
@aethelwulffe this actually does NOT force anything :) maybe you didn't understand well the code (maybe I'm wrong) but this only creates a lowercase version of the argument meaning that I am eliminating the case sensitivity. In other words, you can call "jQuery" "JQUERY" "jquery" "jquErY" and it's all the same. The goal here was to prevent possible bugs or doubts when using the function. 

PS: I'm not a linux user (I wish I were) and I don't know if there are known issues with Linux about this.

Reply:
The library array is used to complete a URL string.  That means that you can pass any library you like into it, and if it is jquery, it completes the URL string with the value.
  If the string value does not exactly match the actual file directory and name (case sensitive) it will break.  By default, Windows file systems do not use case sensitive names, though you can configure that.  By default Linux is exactly the opposite unless you use Darwin or another file system tweak.  Removing hyphens will also kill it.